Doc/library/os.rst

```diff
@@ -1659,6 +1659,27 @@ or `the MSDN <https://msdn.microsoft.com/en-us/library/z0kc8e3z.aspx>`_ on Windo
       :exc:`InterruptedError` exception (see :pep:`475` for the rationale).
 
 
+.. function:: readinto(fd, buffer, /)
+
+   Read from a file descriptor *fd* into a mutable
+   :ref:`buffer protocol <bufferobjects>` *buffer*.
+
+   The *buffer* should be mutable and :term:`bytes-like <bytes-like object>`. On
+   success, returns the number of bytes read. Less bytes may be read than the
+   size of the buffer. Will retry the underlying system call when interrupted by
+   a signal. For other errors, the system call will not be retried.
+
+   .. note::
+
+      This function is intended for low-level I/O and must be applied to a file
+      descriptor as returned by :func:`os.open` or :func:`pipe`.  To read a
+      "file object" returned by the built-in function :func:`open` or by
+      :func:`popen` or :func:`fdopen`, or :data:`sys.stdin`, use its
+      :meth:`~file.readinto` or :meth:`~file.read`.
```

Comment:
```suggestion
      "file object" *fp* returned by the built-in function :func:`open`,
      by :func:`popen` or :func:`fdopen`, or by :data:`sys.stdin`, use
      :meth:`fp.readinto() <file.readinto>` or :meth:`fp.read() <file.read>`
      instead.
```

Reply:
This note is copied from `os.read` just above changing `~file.read` -> `~file.readinto` (Should probably update both..., but avoiding changes outside of "add `os.readinto`" on this PR). The function doesn't take a `fp` so it feels a little weird to do *fp* to me. *fd* also definitely feels wrong as the convention seems to be `fp` is file object, `fd` is int. Definitely will make changes, not sure what. Maybe *file* (which also means new to this people don't need to know fd or fp as much?)

Reply:
Oh yes, you can also use `file` but I think I messed up my references. It should be:

```rst
:meth:`file.read() <io.FileIO.read>` 
:meth:`file.readinto <io.FileIO.readinto>`
```

(or whatever io class it should be, just not `file.read` as the reference)

Reply:
I'll point to `BufferedIO.read/readinto` and `TextIO.read/readinto` + `open()` as that's the behaviors people tend to want / expect.